src/megaapi.cpp

```diff
@@ -3937,576 +3937,574 @@ char *MegaApi::getMimeType(const char *extension)
+        {"wml", "text/vnd.wap.wml"},
+        {"wmlc", "application/vnd.wap.wmlc"},
+        {"wmls", "text/vnd.wap.wmlscript"},
+        {"wmlsc", "application/vnd.wap.wmlscriptc"},
+        {"wmp", "video/x-ms-wmp"},
+        {"wmv", "video/x-ms-wmv"},
+        {"wmx", "video/x-ms-wmx"},
+        {"wmz", "application/x-ms-wmz"},
+        {"wpl", "application/vnd.ms-wpl"},
+        {"wps", "application/vnd.ms-works"},
+        {"wri", "application/x-mswrite"},
+        {"wrl", "x-world/x-vrml"},
+        {"wrz", "x-world/x-vrml"},
+        {"wsc", "text/scriptlet"},
+        {"wsdl", "text/xml"},
+        {"wvx", "video/x-ms-wvx"},
+        {"x", "application/directx"},
+        {"xaf", "x-world/x-vrml"},
+        {"xaml", "application/xaml+xml"},
+        {"xap", "application/x-silverlight-app"},
+        {"xbap", "application/x-ms-xbap"},
+        {"xbm", "image/x-xbitmap"},
+        {"xdr", "text/plain"},
+        {"xht", "application/xhtml+xml"},
+        {"xhtml", "application/xhtml+xml"},
+        {"xla", "application/vnd.ms-excel"},
+        {"xlam", "application/vnd.ms-excel.addin.macroEnabled.12"},
+        {"xlc", "application/vnd.ms-excel"},
+        {"xld", "application/vnd.ms-excel"},
+        {"xlk", "application/vnd.ms-excel"},
+        {"xll", "application/vnd.ms-excel"},
+        {"xlm", "application/vnd.ms-excel"},
+        {"xls", "application/vnd.ms-excel"},
+        {"xlsb", "application/vnd.ms-excel.sheet.binary.macroEnabled.12"},
+        {"xlsm", "application/vnd.ms-excel.sheet.macroEnabled.12"},
+        {"xlsx", "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"},
+        {"xlt", "application/vnd.ms-excel"},
+        {"xltm", "application/vnd.ms-excel.template.macroEnabled.12"},
+        {"xltx", "application/vnd.openxmlformats-officedocument.spreadsheetml.template"},
+        {"xlw", "application/vnd.ms-excel"},
+        {"xml", "text/xml"},
+        {"xmta", "application/xml"},
+        {"xof", "x-world/x-vrml"},
+        {"XOML", "text/plain"},
+        {"xpm", "image/x-xpixmap"},
+        {"xps", "application/vnd.ms-xpsdocument"},
+        {"xrm-ms", "text/xml"},
+        {"xsc", "application/xml"},
+        {"xsd", "text/xml"},
+        {"xsf", "text/xml"},
+        {"xsl", "text/xml"},
+        {"xslt", "text/xml"},
+        {"xsn", "application/octet-stream"},
+        {"xss", "application/xml"},
+        {"xtp", "application/octet-stream"},
+        {"xwd", "image/x-xwindowdump"},
+        {"z", "application/x-compress"},
+        {"zip", "application/x-zip-compressed"}
+    };
 
     string key = extension;
-    std::transform(key.begin(), key.end(), key.begin(), ::tolower);
-    map<string, string>::iterator it = mimeMap->find(key);
-    if (it == mimeMap->end())
+    std::transform(key.begin(), key.end(), key.begin(), [](char c) { return static_cast<char>(::tolower(c)); });
+    map<string, string>::iterator it = mimeMap.find(key);
+    if (it == mimeMap.end())
     {
         return NULL;
```

Comment:
return `nullptr`?

Reply:
That part worked fine, so I left it alone - but since it's come up in review I've switched it now, and made the returns much more compact also.